pkgs/build-support/docker/default.nix

```diff
@@ -801,20 +789,9 @@ rec {
         echo "Warning: only the database of the deepest Nix layer is loaded."
         echo "         If you want to use nix commands in the container, it would"
         echo "         be better to only have one layer that contains a nix store."
-        # This requires Nix 1.12 or higher
+
         export NIX_REMOTE=local?root=$PWD
-        ${nix}/bin/nix-store --load-db < ${nixRegistration contents}/db.dump
-
-        # We fill the store in order to run the 'verify' command that
-        # generates hash and size of output paths.
-        # Note when Nix 1.12 is be the stable one, the database dump
-        # generated by the exportReferencesGraph function will
-        # contains sha and size. See
-        # https://github.com/NixOS/nix/commit/c2b0d8749f7e77afc1c4b3e8dd36b7ee9720af4a
-        storePaths=$(cat ${nixRegistration contents}/storePaths)
-        echo "Copying everything to /nix/store (will take a while)..."
-        cp -prd $storePaths nix/store/
-        ${nix}/bin/nix-store --verify --check-contents
+        ${nix}/bin/nix-store --load-db < ${closureInfo {rootPaths = contents;}}/registration
```

Comment:
`contents` can be a list of derivations. It can also be one single derivation that defines the docker image layer.
Setting `rootPaths` directly to `contents` can break some builds. See this: https://github.com/NixOS/nixpkgs/blob/2f9fc897eb305abee1206061a20f65c12243b0b4/pkgs/build-support/docker/default.nix#L268

Reply:
Would have been nice to have typed Nix here :)
I fixed this. Thanks!